src/v/datalake/translation/deps.cc

```diff
@@ -153,6 +154,63 @@ coordinator_api::make_default_coordinator_api(coordinator::frontend& frontend) {
     return std::make_unique<default_coordinator_api>(frontend);
 }
 
+namespace {
+std::optional<kafka::offset> calculate_max_offset_for_translation(
+  const kafka::partition_proxy& partition_proxy) {
+    // We factor in LSO to ensure only committed transactional batches are
+    // translated.
+    auto lso = partition_proxy.last_stable_offset();
+    if (lso.has_error()) {
+        return std::nullopt;
+    }
+    return kafka::prev_offset(model::offset_cast(lso.value()));
+}
+std::chrono::milliseconds calculate_target_lag(
+  const cluster::topic_table& topics, const model::ntp& ntp) {
+    // todo: In addition to integrating with config subsystem, an additional
+    // task that needs to be done is to keep track of translation intervals
+    // centrally some place (coordinator?). This is needed so that we do not
+    // lose the interval boundaries due to a crash. For example imagine
+    // keeping it in memory and target lag is set to 1d and the leadership
+    // transfer happens after half a day, we cannot just reset the interval
+    // window on the new leader. That probably works for smaller lag values
```

Comment:
Think this was moved from earlier version of the code.. AFAICT this is not needed anymore?